crates/ruff_python_semantic/src/model.rs

```diff
@@ -1247,6 +1247,28 @@ impl<'a> SemanticModel<'a> {
             .nth(1)
     }
 
+    pub fn previous_statement(&self, stmt: &'a Stmt) -> Option<&Stmt> {
+        self.previous_statements(stmt)?.next()
+    }
```

Comment:
I needed something like this function, not sure if there is an existing better way?

Reply:
That makes sense. I think we can make it more efficient with https://github.com/astral-sh/ruff/pull/13895
which reduces an extra find and collect